Refactored the switcher extension to get all deps via injection

thanosp opened the pull request:
- Fixes #200 
- Also fixed some deprecation issues with phpunit

The main problem that remains is that I don't see any real functional test to check the integration with symfony. 
So basically apart from the unit test nothing else broke from the change. Configuration defaults are untested. I can remove an argument from the service definitions and all tests still pass. So that should probably be a new issue.

dbu commented on Tests/LocaleGuesser/LocaleGuesserManagerTest.php:
```diff
@@ -137,7 +137,9 @@ private function getRequestWithoutLocaleQuery()
     private function getGuesserMock()
     {
-        $mock = $this->getMockBuilder('Lunetics\LocaleBundle\LocaleGuesser\LocaleGuesserInterface')->disableOriginalConstructor()->getMock();
+        $mock = $this->getMockBuilder('Lunetics\LocaleBundle\LocaleGuesser\LocaleGuesserInterface')
```
just createMock would be enough, it disables the constructor automatically

dbu commented on Twig/Extension/LocaleSwitcherExtension.php:
```diff
@@ -58,25 +105,24 @@ public function getName()
-        return $this->container->get('lunetics_locale.switcher_helper')->renderSwitch($infos, $template);
+        $showCurrentLocale = $this->showCurrentLocaleParam;
+        $useController = $this->useControllerParam;
+        $allowedLocales = $this->allowedLocalesProvider->getAllowedLocales();
```
we should inline these assignments into the code below directly, i think

thanosp replied:
Yeah indeed. Can I also use short hand array notation yet btw?

thanosp commented:
Replaced class strings with `::class` calls where appropriate.
Replaced mockBuilder calls with createMock.
I've tested this with symfony 3.4 only manually.

marcoreni commented:
Any update about this?

dbu commented:
i hope @lunetics can look at this sometimes. i follow this repository but am not a maintainer of the bundle...

fschaeffer commented:
Any news about this one? I am also at rewriting an application and try to fix as many deprecations as possible. Only these two are left in my profiler ...

marcoreni commented:
I think this is a duplicate of https://github.com/lunetics/LocaleBundle/pull/195 that has been merged & released in 2.6.0

dbu commented:
Its not exactly a duplicate. #195 did the minimum to see this bundle run in symfony 4. This one is a proper refactoring to be better at following best practices. It also includes a cleanup of the tests.

Its currently in conflict with master and needs a rebase.

And it should probably bump the version to 3, there are changes in the constructor of non-final twig extension, so if somebody extended any of the classes or defined their own services with them, things would break. (in version 3, this class should be marked as both `@internal` and be `final`, then we can BC break as much as we want to).

Somebody up to rebase this?
